packages/frontend/src/components/create-campaign/form/kpi-step/index.tsx

```diff
@@ -0,0 +1,320 @@
+
+    const totalRewardsUsdAmount = useMemo(() => {
+        if (!rewards) return 0;
+        let total = 0;
+        for (const reward of rewards) {
+            if (!reward.amount.usdValue) return 0;
+            total += reward.amount.usdValue;
+        }
+        return total;
+    }, [rewards]);
+
+    // this hooks is used to disable and close the step when
+    // the kpi specification gets disabled, after the campaign creation
+    useEffect(() => {
+        if (enabled && !!prevKpiSpecification && !kpiSpecification)
+            setEnabled(false);
+    }, [enabled, kpiSpecification, prevKpiSpecification]);
+
+    useEffect(() => {
+        if (!feedback) return;
+
+        const timeout = setTimeout(() => {
+            setFeedback(false);
+        }, 1500);
+
+        return () => clearTimeout(timeout);
+    }, [feedback]);
+
+    useEffect(() => {
+        if (enabled) return;
+        if (kpiSpecification) onKpiChange({ kpiSpecification: undefined });
+
+        setMinimumPayoutPercentage(0);
+        setLowerUsdTargetRaw(undefined);
+        setUpperUsdTargetRaw(undefined);
+        setBoundsError("");
+    }, [enabled, kpiSpecification, onKpiChange]);
+
+    useEffect(() => {
+        if (!lowerUsdTargetRaw && !upperUsdTargetRaw) {
+            setBoundsError("");
+            return;
+        }
+
+        if (
+            (!lowerUsdTargetRaw && upperUsdTargetRaw) ||
+            (!upperUsdTargetRaw && lowerUsdTargetRaw)
+        ) {
+            setBoundsError("errors.missing");
+            return;
+        }
+
+        if (!lowerUsdTargetRaw || !upperUsdTargetRaw) return;
+
+        const { floatValue: lowerUsdTarget } = lowerUsdTargetRaw;
+        const { floatValue: upperUsdTarget } = upperUsdTargetRaw;
+
+        if (
+            lowerUsdTarget !== undefined &&
+            upperUsdTarget !== undefined &&
+            lowerUsdTarget >= upperUsdTarget
+        ) {
+            setBoundsError("errors.malformed");
+            return;
+        }
+
+        setBoundsError("");
+    }, [lowerUsdTargetRaw, upperUsdTargetRaw]);
+
+    useEffect(() => {
+        onError({
+            kpiSpecification: !!boundsError || (enabled && !kpiSpecification),
+        });
+    }, [boundsError, enabled, kpiSpecification, onError]);
+
+    function handleSwitchOnClick() {
+        setEnabled((enabled) => !enabled);
+    }
+
+    function handleMinimumPayoutOnChange(event: ChangeEvent<HTMLInputElement>) {
+        setMinimumPayoutPercentage(Number(event.target.value));
+    }
+
+    const handleOnApply = useCallback(() => {
+        if (
+            lowerUsdTargetRaw?.floatValue === undefined ||
+            upperUsdTargetRaw?.floatValue === undefined
+        )
+            return;
+
+        const { floatValue: lowerUsdTarget } = lowerUsdTargetRaw;
+        const { floatValue: upperUsdTarget } = upperUsdTargetRaw;
+
+        const kpiSpecification: KpiSpecification = {
+            ...(minimumPayoutPercentage > 0
```

Comment:
I'm wondering if it wouldn't be clearer to set this key later on, on demand only if needed, through the use of an if/else

Reply:
Fixed in 0edb2c51d297f15fb54f2bfbba97283afc4c78e6.

Reply:
Same as above, not seeing the fix here

Reply:
Here 5b46f15ee64470f3ce41249f832d5ae8bf50f923.